### PR TITLE
PR for #33: SQL peer review exercise - Simone

### PR DESCRIPTION
**Pull request by sambock27**
Hello @simonepandit, There is one file for you to review as part of this PR:

- [ ] [`patient_demographics`](https://github.com/dchan-lab/practice/blob/issue33_sql_peer_review/build/code/PatientDemographics.sql)

I've assigned you as the reviewer for this PR. Please peer review the file linked above. Use our best practices for SQL code review [page](https://github.com/dchan-lab/dchanlab/wiki/Best-Practices-for-SQL-Code-Review) as a guide. As a reviewer, you aren't supposed to make any changes to the script.

Let me know if you have any questions!

### CODE REVIEW

File: build/code/PatientDemographics.sql [hunk 1]
```diff
@@ -0,0 +1,58 @@
+
```
Comment (simonepandit) [inline build/code/PatientDemographics.sql:1]: There should be a preamble with a description and information about the author/date created.

File: build/code/PatientDemographics.sql [hunk 2]
```diff
@@ -0,0 +1,58 @@
+
+USE [ORD_Chan_201606104D]
+GO
+
+-- Patient Race
+IF Object_Id('tempdb..#PatientRace') IS NOT NULL
+	DROP TABLE #PatientRace
```
Comment (simonepandit) [inline build/code/PatientDemographics.sql:7]: Does not appear to follow the style guide for indentations (don't need the additional indentation)

File: build/code/PatientDemographics.sql [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+USE [ORD_Chan_201606104D]
+GO
+
+-- Patient Race
+IF Object_Id('tempdb..#PatientRace') IS NOT NULL
+	DROP TABLE #PatientRace
+
+SELECT *
+INTO #PatientRace
+	FROM [Src].[PatSub_PatientRace]
```
Comment (simonepandit) [inline build/code/PatientDemographics.sql:11]: Don't need additional indentation here

File: build/code/PatientDemographics.sql [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	MAX(E.Ethnicity_Hispanic) AS Ethnicity_Hispanic
+INTO [Dflt].[NonVA_PatientDemographics_15]
+FROM (
+	SELECT DISTINCT ScrSSN
+	FROM [Dflt].[NonVA_P1_Ambulances_15]
+) AS A
+INNER JOIN [Src].[SPatient_SPatient] AS CX
+	ON CONVERT(int,A.ScrSSN) = CONVERT(int,CX.ScrSSN)
+LEFT JOIN #PatientEthnicity AS E
+ON CX.PatientSID = E.PatientSID
```
Comment (simonepandit) [inline build/code/PatientDemographics.sql:48]: Need to have an indent here, as per style guide

File: build/code/PatientDemographics.sql [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+LEFT JOIN #PatientRace AS R
+	ON CX.PatientSID = R.PatientSID
+LEFT JOIN [Src].[VitalStatus_Mini] AS V
+	ON CX.ScrSSN=V.SCRSSN
+LEFT JOIN [Src].[Inpat_Inpatient] AS I
+	ON R.PatientSID = I.PatientSID
+WHERE CX.ScrSSN IS NOT NULL AND NOT CX.ScrSSN = '*Unknown at this time*' AND NOT CX.scrSSN = '*Missing*'
```
Comment (simonepandit) [inline build/code/PatientDemographics.sql:55]: Line 55 Should be split across multiple lines and subsequent lines should be indented

File: build/code/PatientDemographics.sql [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+INTO #PatientRace
+	FROM [Src].[PatSub_PatientRace]
+GROUP BY PatientSID;
+
+
+-- Patient Ethnicity
+IF Object_Id('tempdb..#PatientEthnicity') IS NOT NULL
+	DROP TABLE #PatientEthnicity
+
+SELECT PatientSID, MAX(CASE WHEN Ethnicity IN ('HISPANIC OR LATINO') THEN 1 ELSE 0 END) AS Ethnicity_Hispanic
```
Comment (simonepandit) [inline build/code/PatientDemographics.sql:19]: Column name should be on next line

File: build/code/PatientDemographics.sql [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+	-- Vital status
+	MAX(V.DOB) AS DateOfBirth,
+	MAX(V.DOD) AS DateOfDeath,
+	MAX(CASE WHEN V.SEX = 'M' THEN 1 ELSE 0 END) AS Sex_M,
+	-- Race and ethnicity
+	MAX(R.Race_API) AS Race_API, 
+	MAX(R.Race_Black) AS Race_Black,  
+	MAX(R.Race_WhiteNotHispanic) AS Race_WhiteNotHispanic,
+	MAX(E.Ethnicity_Hispanic) AS Ethnicity_Hispanic
+INTO [Dflt].[NonVA_PatientDemographics_15]
```
Comment (simonepandit) [inline build/code/PatientDemographics.sql:40]: There should be a final output table comment here

**Comment by simonepandit** [comment]
@sambock27 I think I've identified some issues, please let me know if I have missed anything.

File: build/code/PatientDemographics.sql (hunk 1 above)
Comment (sambock27) [reply build/code/PatientDemographics.sql:1]: Yes!

File: build/code/PatientDemographics.sql (hunk 3 above)
Comment (sambock27) [reply build/code/PatientDemographics.sql:11]: yeppers

File: build/code/PatientDemographics.sql (hunk 4 above)
Comment (sambock27) [reply build/code/PatientDemographics.sql:48]: yesh

File: build/code/PatientDemographics.sql (hunk 5 above)
Comment (sambock27) [reply build/code/PatientDemographics.sql:55]: mhm

File: build/code/PatientDemographics.sql (hunk 6 above)
Comment (sambock27) [reply build/code/PatientDemographics.sql:19]: yes yes YES

File: build/code/PatientDemographics.sql (hunk 7 above)
Comment (sambock27) [reply build/code/PatientDemographics.sql:40]: What do you mean here? If you mean that there should be a comment stating at which level the data is unique then you would be exactly right.

**Comment by sambock27** [comment]
Thanks for your work on this, @simonepandit. Well done! You identified many of the shortcomings of the code. Below are a few others:

 - The code pulls inpatient data, but that is unneeded because we never save variables from that table
 - there is a compression step missing from the end

That's all! Let me know if you have any questions. Else, feel free to close this issue (without merging to main or deleting this branch)!

**Pull request closed without merging**